src/celmodel/modelfile.cpp

```diff
@@ -2127,251 +1635,332 @@ BinaryModelLoader::loadVertices(const Mesh::VertexDescription& vertexDesc,
+    {
+        if (!writeGroup(*mesh.getGroup(groupIndex))) { return false; }
+    }
 
-    writeToken(out, CMOD_EndMesh);
+    return writeToken(out, CMOD_EndMesh);
 }
 
 
-void
+bool
 BinaryModelWriter::writeVertices(const void* vertexData,
                                  unsigned int nVertices,
                                  unsigned int stride,
                                  const Mesh::VertexDescription& desc)
 {
     const auto* vertex = reinterpret_cast<const char*>(vertexData);
 
-    writeToken(out, CMOD_Vertices);
-    writeUint32(out, nVertices);
+    if (!writeToken(out, CMOD_Vertices) || !celutil::writeLE<std::uint32_t>(out, nVertices))
+    {
+        return false;
+    }
 
     for (unsigned int i = 0; i < nVertices; i++, vertex += stride)
     {
         for (unsigned int attr = 0; attr < desc.nAttributes; attr++)
         {
             const char* cdata = vertex + desc.attributes[attr].offset;
-            const auto* fdata = reinterpret_cast<const float*>(cdata);
+            float fdata[4];
 
+            bool result;
             switch (desc.attributes[attr].format)
             {
             case Mesh::Float1:
-                writeFloat(out, fdata[0]);
+                std::memcpy(fdata, cdata, sizeof(float));
+                result = celutil::writeLE<float>(out, fdata[0]);
                 break;
             case Mesh::Float2:
-                writeFloat(out, fdata[0]);
-                writeFloat(out, fdata[1]);
+                std::memcpy(fdata, cdata, sizeof(float) * 2);
+                result = celutil::writeLE<float>(out, fdata[0])
+                    && celutil::writeLE<float>(out, fdata[1]);
                 break;
             case Mesh::Float3:
-                writeFloat(out, fdata[0]);
-                writeFloat(out, fdata[1]);
-                writeFloat(out, fdata[2]);
+                std::memcpy(fdata, cdata, sizeof(float) * 3);
+                result = celutil::writeLE<float>(out, fdata[0])
+                    && celutil::writeLE<float>(out, fdata[1])
+                    && celutil::writeLE<float>(out, fdata[2]);
                 break;
             case Mesh::Float4:
-                writeFloat(out, fdata[0]);
-                writeFloat(out, fdata[1]);
-                writeFloat(out, fdata[2]);
-                writeFloat(out, fdata[3]);
+                std::memcpy(fdata, cdata, sizeof(float) * 4);
+                result = celutil::writeLE<float>(out, fdata[0])
+                    && celutil::writeLE<float>(out, fdata[1])
+                    && celutil::writeLE<float>(out, fdata[2])
+                    && celutil::writeLE<float>(out, fdata[3]);
                 break;
             case Mesh::UByte4:
-                out.write(cdata, 4);
+                result = out.write(cdata, 4).good();
                 break;
             default:
                 assert(0);
                 break;
             }
+
+            if (!result)
+            {
```

Comment:
could you use the same style everywhere?

Reply:
What style are you referring to here?

Reply:
there are many places where you do `if (...) { return ...; }`. Here you use multiple lines.

Reply:
Got it. Fixed.